### PR TITLE
Preprocess module and subject substitution

### PR DESCRIPTION
**Pull request by moldot**
To assist spacy in parsing, we substitute the text we know refers to
the subject with a fake string that should be recognized as a noun.
This is done in the preprocess module. Also add the
reverse_substitution function in postprocess to get the original string
back

Example:
```
get_assertion_definitions(u"Born to Run is the studio album by the American singer-songwriter Bruce Springsteen.", subject=[u'Born to Run']. get_rep=False)
```

Internally, the sentence will be turned into:
```
POND_NUMBER_0 is the studio album by the American singer-songwriter Bruce Springsteen.
```

The definition tree in an intermediate step would be:
```
[subject: POND_NUMBER_0 [def: album [comp: studio] ... ]]
```

After postprocess with reverse substitution, the tree finally becomes
```
[subject: Born to Run [def: album [comp: studio] ... ]]
```

Unittest has been updated. New unittest class is added to test the substitute function.

### CODE REVIEW

File: whoami/preprocess.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+
+def get_ners_dict(doc):
+    ners = {}
+    for ent in doc.ents:
+        ners[ent.root] = ent.text
+    return ners
+
+
+def get_substitution_str(idx):
```
Comment (alvaromorales) [inline whoami/preprocess.py]: add a comment here that explains the (rather unusual) choice of `POND_NUMBER_X`

**Comment by alvaromorales** [comment]
Looks great! Special congrats for achieving 100% test coverage in the changes you introduced.

For now, we'll only be able to provide the article title as a synonym. This will for for cases like "Born to Run", but not for Bill Clinton where we should pass in "William Jefferson 'Bill' Clinton".

File: whoami/preprocess.py (hunk 1 above)
Comment (moldot) [inline whoami/preprocess.py]: lol I was a little too excited about sneaking my name in there. Actually SUBJECT_ID_n works, too, and might be more understandable. Will add comment saying that we need something that looks like nouns.

Comment (alvaromorales) [inline whoami/preprocess.py]: You said that spaCy sometimes will attempt to "understand" the morphology of fake nouns, e.g. -ED suffix. Are you sure `SUBJECT_ID_n` will always get interpreted as a noun?

Comment (moldot) [inline whoami/preprocess.py]: As far as I have tried, yes.

**Comment by alvaromorales** [comment]
lgtm :rocket:

**Pull request merged**